dapr/proto/runtime/v1/dapr.proto

```diff
@@ -287,6 +290,62 @@ message PublishEventRequest {
   map<string, string> metadata = 5;
 }
 
+// BulkPublishRequest is the message to bulk publish events to pubsub topic
+message BulkPublishRequest {
+  // The name of the pubsub component
+  string pubsub_name = 1;
+
+  // The pubsub topic
+  string topic = 2;
+
+  // The entries which contain the individual events and associated details to be published
+  repeated BulkPublishRequestEntry entries = 3;
+
+  // The request level metadata passing to to the pubsub components
+  map<string, string> metadata = 4;
+}
+
+// BulkPublishRequestEntry is the message containing the event to be bulk published
+message BulkPublishRequestEntry {
+
+  // The event which will be pulished to the topic
+  bytes event = 1;
+
+  // The content type for the event
+  string content_type = 2;
+
+  // The event level metadata passing to the pubsub component
+  map<string, string> metadata = 3;
+
+  // The request scoped unique ID referring to this message. Used to map status in response
+  string entryID = 4;
+}
+
+// BulkPublishResponse is the message returned from a BulkPublishEvent call
+message BulkPublishResponse {
+  // The publish statuses for different events in the BulkPublishEvent call
+  repeated BulkPublishResponseEntry statuses = 1;
+}
+
+// BulkPublishResponseEntry is the message mapping response status for each event in the BulkPublishEvent call
+message BulkPublishResponseEntry {
+
+  enum Status {
+    SUCCESS = 0;
+    FAILED = 1;
+  }
+
+  // The response scoped unique ID referring to this message
+  string entryID = 1;
```

Comment:
Ditto

Reply:
Raised https://github.com/dapr/components-contrib/pull/2141 and https://github.com/dapr/dapr/pull/5285 for this.